activestorage/lib/active_storage/attached/changes/create_many.rb

```diff
@@ -33,7 +33,13 @@ def subchanges
       end
 
       def build_subchange_from(attachable)
-        ActiveStorage::Attached::Changes::CreateOneOfMany.new(name, record, attachable)
+        ActiveStorage::Attached::Changes::CreateOneOfMany.new(name, record, attachable, previous_attachments)
+      end
+
+      # Using a direct query instead of `record.public_send("#{name}_attachments")` because of a race condition
+      # where some attachments were missing due to them being memoized in a previous stage
+      def previous_attachments
```

Comment:
Instead of forcing to load the data from the DB by default what about rescuing the `UniqueViolation: ERROR` in the right part of the code and only in such situation loading the data from the DB?

My point is that this change will introduce a performance degradation by default just to handle the rough edge case, rather take care of the rough edge case explicitly.

Reply:
We thought about it and it actually fixes the issue, but it doesn't seem like a long-term solution. We could also cause issues since this rescuing the error might cause other things to fail silently. In general, I think it's better to kill the error from the root rather than rescue its consequences.

I also think this error might be happening on other cases and we are not adding such a big performance degradation, it's actually the same if you use just one `attach` and then 1 more per attach.

Consider this example:
```ruby
user.documents.attach(some_docs1)
user.documents.attach(some_docs2)
user.documents.attach(some_docs3)
user.documents.attach(some_docs4)
```

So before this would cause `X` queries to be generated, now it would generate `X` + 3 since it's not using the memorized attachments. I don't think it's such a big deal considering this fixes a race condition which can cause pretty nasty bugs. That said, it'd be cool to reduce that as well but it hasn't been easy 😅

Reply:
I don't have any strong opinions here but one thought was this process is very IO bound, so we probably don't have to worry about generating a few extra DB queries. 🤔